File: HorseRacing.Contracts/Models/User/Requests/GetAccountBalance/GetAccountBalanceRequest.cs
using HorseRacing.Contracts.Base.Requests;

namespace HorseRacing.Contracts.Models.User.Requests.GetAccountBalance
{
    public class GetAccountBalanceRequest : BaseRequest<GetAccountBalanceRequestDto>
    {
        public GetAccountBalanceRequest() : base() { }
        public GetAccountBalanceRequest(GetAccountBalanceRequestDto data) : base(data) { }
    }
}

File: HorseRacing.Web/src/interfaces/api/contracts/model/user/requests/get-account-balance/get-account-balance-request.ts
import type { BaseRequest } from '~/interfaces/api/contracts/base/requests/base-request';
import type { GetAccountBalanceRequestDto } from './get-account-balance-request-dto';

export class GetAccountBalanceRequest implements BaseRequest<GetAccountBalanceRequestDto> {
    Data: GetAccountBalanceRequestDto;

    constructor(data?: GetAccountBalanceRequestDto) {
        this.Data = data || { UserId: '' };
    }
}
